philipmeneghini/InstaConnect
# Search Bar

**Backend**
Created two new search endpoints specifically designed for searching for `UserModel` and `ContentModel` in the database

- `/api/Search/User`: takes a string as a query parameter which is split by whitespaces and searched for users in the database using regex
- `/api/Search/Content`: takes a string as a query parameter which is split by whitespaces and searches for contents in the databse using regex

**Frontend**
Search bar placed in the `Header` component with a dropdown menu for search results
![SearchBar](https://github.com/philipmeneghini/InstaConnect/assets/63440066/06656c5a-6f4c-4e98-ba0e-00182ff36cd8)

**Testing**

- Check that endpoints both work as suspected when running different search queries
- Test the search results in the frontend search bar to verify that results match and the rendering is correct when hitting 'Enter'
- Click on the search results to navigate to the item
